crates/tui/src/main.rs

```diff
@@ -0,0 +1,389 @@
+#![feature(let_chains)]
+
+use std::{
+    fs::OpenOptions,
+    path::PathBuf,
+    sync::{RwLock, RwLockWriteGuard},
+    time::Duration,
+};
+
+use alloy::signers::{k256::ecdsa::SigningKey, local::LocalSigner};
+use crossterm::event::{Event, EventStream, KeyCode, KeyEvent};
+use eyre::OptionExt;
+use futures::StreamExt;
+use ratatui::{
+    layout::{Constraint, Layout},
+    prelude::{Buffer, Rect},
+    style::{Color, Style, Stylize},
+    text::Text,
+    widgets::{Block, Borders, FrameExt, List, ListState, Paragraph, StatefulWidget, Widget},
+    DefaultTerminal, Frame,
+};
+use tokio::sync::mpsc::{self};
+use tracing_subscriber::EnvFilter;
+
+/// Returns the base config directory for nexum. It also creates the directory
+/// if it doesn't exist yet.
+fn config_dir() -> eyre::Result<PathBuf> {
+    let dir = std::env::home_dir()
+        .ok_or_eyre("home directory not found")?
+        .join(".nexum");
```

Comment:
```suggestion
        .join(".nxm");
```

Would suggest that for filesystem naming we use `nxm` as an abbreviation?